sources/packages/api/src/route-controllers/institution/institution.controller.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Post,
+  UnprocessableEntityException,
+} from "@nestjs/common";
+
+import { Public } from "../../auth/decorators/public.decorator";
+
+import { InstitutionService, UserService } from "../../services";
+import { CreateInstitutionDto } from "./models/institution.dto";
+import { UserToken } from "../../auth/decorators/userToken.decorator";
+import { IUserToken } from "../../auth/userToken.interface";
+import BaseController from "../BaseController";
+
+@Controller("institution")
+export class InstitutionController extends BaseController {
+  constructor(
+    private readonly userService: UserService,
+    private readonly institutionService: InstitutionService,
+  ) {
+    super();
+  }
+
+  @Post()
+  async create(
+    @Body() payload: CreateInstitutionDto,
+    @UserToken() userToken: IUserToken,
+  ) {
+    // Check user exists or not
+    const existingUser = await this.userService.getUser(userToken.userName);
+    if (existingUser) {
+      throw new UnprocessableEntityException("Institution User already exists");
+    }
+
+    // Save student
+    return this.institutionService.createInstitution(userToken, payload);
```

Comment:
The read-only data that is retrieved from BCeID Web Service should not be expected to be received from the API client.
User first name, last name, institution business GUID, and institution operating name should be retrieved during the institution creation.

Reply:
I dint get this comment - lets discuss

Reply:
The data that is received from BCeID Web Service and is displayed to the client in the UI, should again be retrieved during the user creation in the API. We should only allow the client to update the data that he is able to edit.
The source of truth for the read-only data is BCeID Web Service and in my option, the data is going to be persisted to our DB should come from BCeID Web Service and not from the UI.

Reply:
Done